### PR TITLE
Dfoxpowell/mixpanel cookie cleanup script

### PR DESCRIPTION
**Pull request by dtothefp**
This PR addresses keeping the mixpanel cookie under control https://optimizely.atlassian.net/browse/MKT-1195

- refactored grunt/config task to make dependency ordering more clear
- add the new method for trimming the mixpanel cookie into concat task for both main bundle and om bundle
- calling the new method inside the Segment ready function in global.js
- refactored mixpanel code to be more defensive for older browsers

### To Test

set a breakpoint in staging at the `trimMixpanelCookie` method. The method will gather the experiment names but won't actually remove anything because in staging `mixpane.cookie.props` does not contain any keys with the string `Experiments: `

screen shot 2015-02-19 at 2 15 17 pm

#### sample output on production
screen shot 2015-02-19 at 3 04 35 pm

### CODE REVIEW

File: grunt/concat.js [hunk 1]
```diff
@@ -1,8 +1,64 @@
...
+  'utils/form_helper_factory.js',
+  'utils/form_helpers/*.js',
+  'utils/trim_url.js',
+  'global.js',
+  'components/*.js',
+  'services/*.js',
+  '!services/user_state.js'
+];
+
 module.exports = function(grunt, options){
```
Comment (dtothefp) [inline grunt/concat.js]: @kylerush it was bugging me that it was hard to envision our dependency order, I think this makes it more clear.

File: website-guts/assets/js/global.js [hunk 2]
```diff
@@ -520,3 +520,6 @@ w.optly_q.push([function(){
 }]);
 
 w.optly.mrkt.formHadError = false;
+
+//call the utility function to unregister archived experiments from the mixpanel cookie
+window.analytics.ready(window.optly.mrkt.utils.trimMixpanelCookie);
```
Comment (dtothefp) [inline website-guts/assets/js/global.js]: @kylerush passing the new method to the segment ready function

File: website-guts/assets/js/global.js [hunk 3]
```diff
@@ -521,6 +521,9 @@ w.optly_q.push([function(){
 
 w.optly.mrkt.formHadError = false;
 
+//call the utility function to unregister archived experiments from the mixpanel cookie
+window.analytics.ready(window.optly.mrkt.utils.trimMixpanelCookie);
```
Comment (kylerush) [inline website-guts/assets/js/global.js]: @dtothefp can you the `window` to `w`

**Comment by kylerush** [comment]
LGTM

**Pull request merged**